src/util/clip.js

```diff
@@ -19,12 +21,112 @@ export function clip(val, low, high) {
  *
  * @param  {Object} point The point to clip
  * @param  {Object} box   The bounding box to clip to.
- * @returns {Object}       The clipped point.
+ * @returns {void}
  */
 export function clipToBox(point, box) {
-  // Clip an {x, y} point to a box of size {width, height}
-  point.x = clip(point.x, 0, box.width);
-  point.y = clip(point.y, 0, box.height);
+  // Clip an {x, y} point to a box {top, left, width, height}
+  const left = box.left || 0;
+  const top = box.top || 0;
+
+  point.x = clip(point.x, left, left + box.width);
+  point.y = clip(point.y, top, top + box.height);
+}
+
+/**
+ * Returns a new bounding box of rotated text box, relative to the pixel
+ * coordinate system. It will get the coordinate of the 4 points of the rotated
+ * text box and calculate the AABB (axis-aligned bounding box - lower and upper
+ * boundaries for `x` and `y` axes).
```

Comment:
I don't understand the documentation here, what is rotated? text box is rotated?